packages/aws-cdk-lib/aws-batch/lib/managed-compute-environment.ts

```diff
@@ -88,7 +88,12 @@ export interface IManagedComputeEnvironment extends IComputeEnvironment, ec2.ICo
    *
    * If you specify a specific AMI, this property will be ignored.
    *
-   * @default true
+   * Note: the CDK will never set this value by default, `false` will set by CFN.
+   * This is to avoid a deployment failure that occurs when this value is set.
+   *
+   * @see https://github.com/aws/aws-cdk/issues/27054
+   *
+   * @default false
```

Comment:
I might be missing something here, but the linked issue makes it sound like setting this prop to `true` can break the deployment. 

Can we fix the underlying issue before stabilization?

Reply:
The underlying issue is in CFN, and this is the fix. The problem is that whenever a ComputeEnvironment is created, the CFN handler first calls CreateCE, and then UpdateCE. The issue with this is that CreateCE does not set certain properties (in this case, this prop). UpdateCE is given these ignored properties, but UpdateCE sees that this property has changed; and, as the error indicates, these properties are not allowed to be changed.

CDK's default value is causing this issue by passing it in as `true` by default. The same issue would occur if it was `false`. This only occurs with `Fargate`, but the service team suggested the value not be set by default anywhere, so that is what this does. 

The issue for this is really a needs-cfn, I'll update the issue.